core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -220,6 +220,21 @@ private[spark] class Executor(
 
   heartbeater.start()
 
+  private val appStartTime = conf.getLong("spark.app.startTime", 0)
+
+  // Jars and files specified by spark.jars and spark.files.
```

Comment:
it would be nice to expand these comment stating why we do this specially and it has to be before the plugins run

Reply:
nit: Given both are identical code blocks, with only the key being different - refactor it ?

Reply:
Thanks. I've refactored and expanded the comment.